testsuite/peptide_tests.py

```diff
@@ -16,87 +16,72 @@
+tasks = [
+  "K"*5+"D"*5, # K_5-D_5 case
+  "E"*5+"H"*5, # E_5-H_5 case
+  "nDSHAKRHHGYKRKFHEKHHSHRGYc", # histatin-5 case, slow simulation
+]
+mode = "test"
 
-def run_peptide_test(script_path,test_pH_values,sequence,rtol,atol,mode="test"):
+def kernel(sequence):
     """
     Runs a set of tests for a given peptide sequence.
 
     Args:
-        script_path(`str`): Path to the script to run the test.
-        test_pH_values(`lst`): List of pH values to be tested.
         sequence(`str`): Amino acid sequence of the peptide.
     """
-    valid_modes=["test","save"]
-    assert mode in valid_modes, f"Mode {mode} not supported, valid modes: {valid_modes}"
-
-    print(f"Running tests for {sequence}")
     with tempfile.TemporaryDirectory() as time_series_path:
         for pH in test_pH_values:
             print(f"pH = {pH}")
-            run_command=[sys.executable, script_path, "--sequence", sequence, "--pH", str(pH), "--mode", "test", "--no_verbose", "--output", time_series_path]
+            run_command=[sys.executable, script_path, "--sequence", sequence,
+                         "--pH", str(pH), "--mode", "test", "--no_verbose",
+                         "--output", time_series_path]
             print(subprocess.list2cmdline(run_command))
             subprocess.check_output(run_command)
         # Analyze all time series
         data=analysis.analyze_time_series(path_to_datafolder=time_series_path)
-        data_path=pmb.get_resource(path="testsuite/peptide_tests_data")
-    if mode == "test":
-        # Get reference test data
-        ref_data=pd.read_csv(data_path+f"/{sequence}.csv", header=[0, 1])
-        # Check charge
-        test_charge=np.sort(data["mean","charge"].to_numpy())
-        ref_charge=np.sort(ref_data["mean","charge"].to_numpy())
-        np.testing.assert_allclose(test_charge, ref_charge, rtol=rtol, atol=atol)
-        # Check rg
-        test_rg=np.sort(data["mean","rg"].to_numpy())
-        ref_rg=np.sort(ref_data["mean","rg"].to_numpy())
-        np.testing.assert_allclose(test_rg, ref_rg, rtol=rtol, atol=atol)
-        print(f"Test for {sequence} was successful")
-    elif mode == "save":
-        # Save data for future testing
-        data.to_csv(f"{data_path}/{sequence}.csv", index=False)
-    else:
-        raise RuntimeError
-
-# Create an instance of pyMBE library
-pmb = pyMBE.pymbe_library(seed=42)
+    return (sequence, data)
 
-script_path=pmb.get_resource("samples/Beyer2024/peptide.py")
-test_pH_values=[3,7,11]
-rtol=0.1 # relative tolerance
-atol=0.5 # absolute tolerance
 
-# Run test for K_5-D_5 case
-sequence="K"*5+"D"*5
+class Test(ut.TestCase):
 
-run_peptide_test(script_path=script_path,
-                    test_pH_values=test_pH_values,
-                    sequence=sequence,
-                    rtol=rtol,
-                    atol=atol)
+    def test_peptide(self):
+        with multiprocessing.Pool(processes=2) as pool:
+            results = dict(pool.map(kernel, tasks, chunksize=2))
```

Comment:
Some of the longer tests now leverage multiprocessing. We won't get better performance with more than 2 processes: for peptides, histatin-5 takes as long as the two short peptides, and for globular proteins, both take the same time. While this doesn't make much of a difference with `make tests -j4` (which is what we have in CI), with `make tests -j8` on laptops that have more than 4 cores, it's noticeably faster, because the CTest scheduler can do load balancing more efficiently. Until now, the two slowest tests took 20 min each, therefore CTest could not run any faster than 20 min. Now they take 12 min resp. 9 min each, which is in line with the third slowest test (15 min). 15 min is a huge improvement compared to the 50 min the old Makefile-based testsuite took.

If we later choose to add more proteins or peptide sequences, the specific order of strings in `tasks` and the value of `chunksize` need to be adapted to optimize load balancing between the two processes. One can increase the number of processes in the Python file, but then the corresponding variable `THREADS` in the CTest config file need to reflect that. I did not introduce multiprocessing in `weak_polyelectrolyte_dialysis_test.py` since we have only 1 molecule, and the pH values have inhomogeneous runtimes, such that with 2 processes, one of the process remains idle for 50 seconds waiting for the other process to complete the simulation at pH 9.